libfwupdplugin/fu-plugin.c

```diff
@@ -507,6 +518,10 @@ fu_plugin_device_add (FuPlugin *self, FuDevice *device)
 		return;
 	}
 
+	/* add to array */
+	fu_plugin_ensure_devices (self);
+	g_ptr_array_add (priv->devices, g_object_ref (device));
```

Comment:
are you missing the matching unref on remove?

Reply:
No, `g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);` speficies the function to use when deallocating or removing items from the array.